filebeat/inputsource/tcp/conn.go

```diff
@@ -71,7 +71,7 @@ type DeadlineReader struct {
 }
 
 // NewDeadlineReader returns a new DeadlineReader
-func NewDeadlineReader(c net.Conn, timeout time.Duration) *DeadlineReader {
+func NewDeadlineReader(c net.Conn, timeout time.Duration) io.Reader {
```

Comment:
Why change the return type here? Prefer to return concrete types, but accept interfaces.